src/Esfa.Vacancy.Register.Api/Mappings/ApprenticeshipMapper.cs

```diff
@@ -62,6 +62,26 @@ public ApprenticeshipVacancy MapToApprenticeshipVacancy(Domain.Entities.Apprenti
             return apprenticeship;
         }
 
+        private WageUnit MapWageUnit(int? wageUnitId)
```

Comment:
Looking at the tests, if null and zero both are expected to map to unspecified, I would have just populated 0 in the mapper and used Enum with values to get correct conversion and avoid this extra coding all together.